### DISCUSSION TITLE
Read dynamic hotspot data inside Blender

### DISCUSSION THREAD

**Discussion by Trolli991** [google/model-viewer#4429] [Q&A]
I am currently trying to read the dynamic hotspot data within Blender using a Python script. I am proceeding as follows:

An Empty object can be moved via the snap to surface option on a mesh object to where the dynamic hotspot should be created later. Via Python BVHTree method I search (via Raycast) for the closest face on the closest object. 
As you know, a dynamic hotspot consists out of 8 values: "glTF mesh, primitive, three vertex indices and three barycentric coordinates to specify a surface point that will animate along with the mesh"

If the closest face was found, the three vertex indexes are read, which are the 3,4 & 5 value of the dynamic hotspot. Next, the positions of the vertices and the empty object in 3d space are used to calculate the barycentric coordinates via mathutils vector and mathutils.geometry.area_tri. The resulting values are 6,7,8.

So everything to get the last 6 values is already there however I can't find a method to get the first two values.

I'm thankful for any help 😅

**Comment by elalish** [comment]
Well, when a glTF is written, all meshes end up in a vector with implicit indices. Same with the primitives inside each mesh (though often there's only one). So you'll need to know something about how Blender writes its glTF to have an association between the mesh in the scene and the mesh index in the output. 

Also, I'm concerned as I would expect barycentric coordinates to always be float values between 0.0 and 1.0, so 6,7,8 looks like a problem.

**Reply by Trolli991** [reply]
Thanks for your fast response. 
I kind of figured that there was no simple solution to this problem, but that I would have to read pages and pages of documentation again. I guess you don't have any experience with Blender either, right? 

Don't worry about the barycentric coordinates. I guess I expressed myself a little wrong. I meant that the values resulting from my Python script result in the sixth, seventh and eighth value of the hotspot data. This works fine and the values are between 0 and 1.

**Reply by elalish** [reply]
No, I don't have much experience with Blender, but I think your feature sounds great. Sorry for the barycentric confusion. :)
